app/loader.cpp

```diff
@@ -263,3 +303,22 @@ QList<QgsExpressionContextScope *> Loader::globalProjectLayerScopes( QgsMapLayer
   scopes << QgsExpressionContextUtils::layerScope( layer );
   return scopes;
 }
+
+void Loader::setActiveLayerFromName( QString layerName ) const
+{
+  mLayersProxyModel.invalidate();
+  mActiveLayer.setActiveLayer(
+    mLayersProxyModel.layerFromName( layerName )
+  );
+}
+
+QString Loader::loadIconFromLayer( QgsMapLayer *layer )
+{
+  if ( !layer )
+    return QString();
+
+  return mLayersProxyModel.data(
+           mLayersProxyModel.index( mLayersProxyModel.indexFromLayer( layer ), 0 ),
+           LayersModel::IconSourceRole
+         ).toString();
```

Comment:
Again, not sure why we need a model here... can't we just check layer's geometry type and return the icon path directly without model?

I think ideally Loader should be free from any model-specific code ...